node/consensus/data/consensus_frames.go

```diff
@@ -21,12 +21,13 @@ import (
 
 const defaultSyncTimeout = 4 * time.Second
 
-func (e *DataClockConsensusEngine) collect(
-	enqueuedFrame *protobufs.ClockFrame,
-) (*protobufs.ClockFrame, error) {
+func (e *DataClockConsensusEngine) syncWithMesh() error {
 	e.logger.Info("collecting vdf proofs")
 
-	latest := enqueuedFrame
```

Comment:
this has the possibility it may be behind and refetch already retrieved frames, we should use the max of enqueued frame or time reel head

Reply:
That's exactly what the change does, right ?

Reply:
The enqueued frame, by the time it reaches collect, _was_ the head at some point. Collecting the head at this point is guaranteed to be at least equal or greater to the enqueued frame.

Reply:
Ah right, makes sense – I was still thinking from the mental model where collect occurred at a prior callsite